blog/_posts/2021-12-2-measuring-dev-env-health.md

```diff
@@ -0,0 +1,211 @@
+---
+slug: "measuring-dev-env-health"
+date: 2021-12-02
+author: nick
+layout: blog
+title: "Measuring Dev Env Health with Honeycomb"
+subtitle: "A tour of the Tilt Honeycomb extension, how it scrapes data, and how you can collect your own metrics"
+description: "A tour of the Tilt Honeycomb extension, how it scrapes data, and how you can collect your own metrics"
+image: "/assets/images/measuring-dev-env-health/cover.jpg"
+tags:
+  - metrics
+  - docker
+  - honeycomb
+---
+
+If you work with containers, and you have friends who also work with containers,
+you probably already know that image build performance can vary WILDLY from
+machine to machine.
+
+Here's a recent conversation from the Tilt Dev slack channel:
+
+---
+
+**Matt**
+I think I'm up to 90 minutes of update-codegen wall time for the day, might have to actually look into what it's doing
+
+**nick**
+is it possible you've got docker cpu or memory very constrained? for me it's about 60sec. curious how long it takes for @milas ?
+
+**Matt**
+I had a vague memory of mounted volume performance being bad on macos and that was what made toast proto slow and had handwavily assumed that was the case here as well
+
+**milas**
+I'm running it now and will see how long it takes - I've given Docker 4 cores + 6GB of RAM (I've got 6/16 physical on my MBP).
+
+**milas**
+make update-codegen  1.37s user 0.62s system 0% cpu 4:55.05 total
+~5 mins, which is what it's been in the past
+
+---
+
+Sure, a build that varies between 60 seconds and 90 minutes is an outlier, but
+it's not an uncommon outlier!
+
+Many Tilt users start to worry about this: are people on the team seeing vastly
+different behavior from their dev environment? How can we catch this?
+
+In this post, I'm going to share some examples of an approach that's worked well
+for us: the health collector sidecar.
+
+I'll show you how to:
+
+1. Scrape the Tilt API to collect the data points you need.
```

Comment:
Maybe give a couple examples of data points, e.g. image build duration or time to resource ready

Reply:
done